src/test/kotlin/no/nav/yrkesskade/saksbehandling/service/BehandlingServiceTest.kt

```diff
@@ -324,4 +338,98 @@ class BehandlingServiceTest : AbstractTest() {
+
+    @Test
+    fun `overfoer behandling til legacy system (opprett journalpostoppgave) - feil saksbehandlinger`() {
+        // given
+        Mockito.`when`(autentisertBruker.preferredUsername).thenReturn("ingen")
+        Mockito.`when`(safClient.hentOppdatertJournalpost(anyString())).thenReturn(okRespons().data)
+        val behandling = behandlingRepository.save(
+            BehandlingEntityFactory.enBehandling("test")
+                .medSak(sak)
+                .medBehandlingstype(Behandlingstype.JOURNALFOERING)
+                .medStatus(Behandlingsstatus.UNDER_BEHANDLING)
+        )
+
+        // when
+        assertThrows<IllegalStateException> {
+            behandlingService.overforBehandlingTilLegacy(behandling.behandlingId, "ingen")
+        }
+    }
+
+    @Test
+    fun `overfoer behandling til legacy system (opprett journalpostoppgave) - ingen journalføring funnet`() {
+        // given
+        Mockito.`when`(autentisertBruker.preferredUsername).thenReturn("test")
+        val behandling = behandlingRepository.save(
+            BehandlingEntityFactory.enBehandling("test")
+                .medSak(sak)
+                .medBehandlingstype(Behandlingstype.JOURNALFOERING)
+                .medStatus(Behandlingsstatus.UNDER_BEHANDLING)
+        )
+
+        // when
+        assertThrows<IllegalStateException> {
+            behandlingService.overforBehandlingTilLegacy(behandling.behandlingId, "ingen")
+        }
+    }
+
+    @Test
+    fun `overfoer behandling til legacy system (opprett journalpostoppgave) - er ikke journalfoering`() {
+        // given
+        Mockito.`when`(autentisertBruker.preferredUsername).thenReturn("test")
+        Mockito.`when`(safClient.hentOppdatertJournalpost(anyString())).thenReturn(okRespons().data)
+        val behandling = behandlingRepository.save(
+            BehandlingEntityFactory.enBehandling("test")
+                .medSak(sak)
+                .medBehandlingstype(Behandlingstype.VEILEDNING)
+                .medStatus(Behandlingsstatus.UNDER_BEHANDLING)
+        )
+
+        // when
+        assertThrows<IllegalStateException> {
+            behandlingService.overforBehandlingTilLegacy(behandling.behandlingId, "ingen")
+        }
+    }
+
+    @Test
+    fun `overfoer behandling til legacy system (opprett journalpostoppgave) - er ferdig`() {
+        // given
+        Mockito.`when`(autentisertBruker.preferredUsername).thenReturn("test")
+        Mockito.`when`(safClient.hentOppdatertJournalpost(anyString())).thenReturn(okRespons().data)
+        val behandling = behandlingRepository.save(
+            BehandlingEntityFactory.enBehandling("test")
+                .medSak(sak)
+                .medBehandlingstype(Behandlingstype.JOURNALFOERING)
+                .medStatus(Behandlingsstatus.FERDIG)
+        )
+
+        // when
+        assertThrows<IllegalStateException> {
+            behandlingService.overforBehandlingTilLegacy(behandling.behandlingId, "ingen")
+        }
+    }
```

Comment:
Burde vi hatt et en test for når den allerede er overført også?

Reply:
Jeg tar bort statusen.